packages/block-editor/src/components/rich-text/index.js

```diff
@@ -113,8 +113,14 @@ export function RichTextWrapper(
 	props = removeNativeProps( props );
 
 	const anchorRef = useRef();
-	const { clientId } = useBlockEditContext();
+	const { clientId, isSelected: isBlockSelected } = useBlockEditContext();
 	const selector = ( select ) => {
+		// Avoid subscribing to the block editor store if the block is not
+		// selected.
+		if ( ! isBlockSelected ) {
+			return { isSelected: false };
+		}
+
```

Comment:
Some of the "if" checks under this can be simplified now no?

Reply:
I don't think so? It checks if the rich text instance is selected, it doesn't involve the block selection check